Update maintainers in README.md

jasonwilliams opened the pull request:
On top of updating the new temporal team i've also updated the maintainers list. I've removed myself from the top as im not as involved as others are (plus i fall under boa developers). 

I've also added @Manishearth to the core maintainers list

nekevss reviewed (approved):
I'm fine with this.

I think it would also make sense to have the general temporal team listed as well.

jasonwilliams commented:
I can add myself back in and then I think that’s the temporal team covered for now, unless I’ve missed someone out

jasonwilliams commented:
Ok I've updated the names in the commit and added back myself and @HalidOdat to the names to reflect the Temporal team plus linked to the Boa team too

HalidOdat commented:
> @HalidOdat has been doing some commits and reviews here and there. Idk, if he had wanted to be included too on the team. But we can always do that in a follow up.

I'd love to be included as well! I've been wanting to get more involved with temporal_rs and dive deeper into the Temporal spec and all things datetime/timezone-related. Looking forward to contributing more! 😄
